Applications/Spire/Source/KeyBindings/KeyBindings.cpp

```diff
@@ -37,30 +30,50 @@ void KeyBindings::reset_binding(const Region& region,
   }
 }
 
-boost::optional<const KeyBindings::Action&>
-    KeyBindings::find_binding(const Region& region,
-    const QKeySequence& sequence) const {
+boost::optional<KeyBindings::Action> KeyBindings::find_binding(
+    const Region& region, const QKeySequence& sequence) const {
   auto i = m_bindings.find(sequence);
   if(i == m_bindings.end()) {
     return boost::none;
   }
-  auto& action = i->Get(region);
-  if(action) {
-    return *action;
-  } else {
-    return boost::none;
-  }
+  return i->Get(region);
 }
 
 KeyBindings::OrderBindingsList
     KeyBindings::build_order_bindings_list() const {
   auto list = OrderBindingsList();
+  std::for_each(m_bindings.constKeyValueBegin(), m_bindings.constKeyValueEnd(),
+    [&] (auto& e) {
+      auto& elements = e.second;
+      for(auto i = elements.Begin(); i != elements.End(); ++i) {
+        auto& action = std::get<1>(*i);
```

Comment:
Prefer to use:

    if(auto& action = std::get<1>(*i)) {
      if(auto order_action = std::get_if(...)) {
        ...
      }
    }